### DISCUSSION TITLE
How to "write" ASN on documents

### DISCUSSION THREAD

**Discussion by Jinzuu** [jonaswinkler/paperless-ng#1641] [Support] [answered]
Hi All,

the official getting started states that 

> "If you need to keep the document, write a running number on the document before scanning,"

I tried actually writing the number on the document, but the OCR retrieves garbage. I had four zeros in the number and every zero was OCRed as something else - but never an actual zero.

Am I doing something wrong in the configuration of the OCR, or is it simply that I should get a stamp and stamp the ASN on the document?

How do you guys solve this?

**Comment by ma651851384** [comment, marked as answer]
Hi Junzuu,
do you expect that paperless-ng automatically detects the ASN? My recommentation:

1. Get the last used ASN from previous document.
2. Write the next ASN (previous number +1) with or without leading zeros on the document.
3. Upload the document to paperless-ng. Open the webpage yet (or later) and open the document-edit page. You will see the content of the document as well as your used ASN
4.  Add manually the ASN number. There is  a "ASN+1" button, to make it easier.

Regards.

**Reply by Jinzuu** [reply]
Yes, I was expecting paperless-ng to automatically detect the ASN. The OCR would detect that number. And thus, I can find the digital document by searching for the ASN.

However, now that I'm writing this, something occured to me: The use-case I had in mind, doesn't really make sense.....
I was thinking about searching the digital document via ASN. But that use-case is actually not important. I can find the digital document by other means, for which I don't need the ASN.
And for the other way around (knowing the digital document, and finding the physical document), a hand-written number suffices, as I can easily identify the number by looking at the digital document.

Thanks, @ma651851384 , so in a way, you helped to figure out the answer :) Thanks.

**Reply by ma651851384** [reply]
You're welcome!
